File: SDKs/googleapis.com/logging/v2/python/src/sdk/models/shared/logbucket.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *

class LogBucketLifecycleStateEnum(str, Enum):
    LIFECYCLE_STATE_UNSPECIFIED = "LIFECYCLE_STATE_UNSPECIFIED"
    ACTIVE = "ACTIVE"
    DELETE_REQUESTED = "DELETE_REQUESTED"


@dataclass_json
@dataclass
class LogBucket:
    r"""LogBucket
    Describes a repository in which log entries are stored.
    """
    
    cmek_settings: Optional[CmekSettings] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cmekSettings') }})
    create_time: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('createTime') }})
    description: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    index_configs: Optional[List[IndexConfig]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('indexConfigs') }})
    lifecycle_state: Optional[LogBucketLifecycleStateEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lifecycleState') }})
    locked: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('locked') }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    restricted_fields: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('restrictedFields') }})
    retention_days: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('retentionDays') }})
    update_time: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('updateTime') }})
    

@dataclass_json
@dataclass
class LogBucketInput:
    r"""LogBucketInput
    Describes a repository in which log entries are stored.
    """
    
    cmek_settings: Optional[CmekSettingsInput] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cmekSettings') }})
    description: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    index_configs: Optional[List[IndexConfigInput]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('indexConfigs') }})
    locked: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('locked') }})
    restricted_fields: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('restrictedFields') }})
    retention_days: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('retentionDays') }})
    

File: SDKs/googleapis.com/logging/v2/typescript/src/sdk/models/shared/logbucket.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CmekSettingsInput } from "./cmeksettings";
import { IndexConfigInput } from "./indexconfig";
import { CmekSettings } from "./cmeksettings";
import { IndexConfig } from "./indexconfig";


export enum LogBucketLifecycleStateEnum {
    LifecycleStateUnspecified = "LIFECYCLE_STATE_UNSPECIFIED",
    Active = "ACTIVE",
    DeleteRequested = "DELETE_REQUESTED"
}


// LogBucketInput
/** 
 * Describes a repository in which log entries are stored.
**/
export class LogBucketInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cmekSettings" })
  cmekSettings?: CmekSettingsInput;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=indexConfigs", elemType: IndexConfigInput })
  indexConfigs?: IndexConfigInput[];

  @SpeakeasyMetadata({ data: "json, name=locked" })
  locked?: boolean;

  @SpeakeasyMetadata({ data: "json, name=restrictedFields" })
  restrictedFields?: string[];

  @SpeakeasyMetadata({ data: "json, name=retentionDays" })
  retentionDays?: number;
}


// LogBucket
/** 
 * Describes a repository in which log entries are stored.
**/
export class LogBucket extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cmekSettings" })
  cmekSettings?: CmekSettings;

  @SpeakeasyMetadata({ data: "json, name=createTime" })
  createTime?: string;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=indexConfigs", elemType: IndexConfig })
  indexConfigs?: IndexConfig[];

  @SpeakeasyMetadata({ data: "json, name=lifecycleState" })
  lifecycleState?: LogBucketLifecycleStateEnum;

  @SpeakeasyMetadata({ data: "json, name=locked" })
  locked?: boolean;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=restrictedFields" })
  restrictedFields?: string[];

  @SpeakeasyMetadata({ data: "json, name=retentionDays" })
  retentionDays?: number;

  @SpeakeasyMetadata({ data: "json, name=updateTime" })
  updateTime?: string;
}
